src/sections/node/tickets.tsx

```diff
@@ -44,28 +44,13 @@ function TicketsPage() {
   };
 
   const handleRedeemAllTickets = () => {
-    set_redeeming(true);
     dispatch(
       actionsAsync.redeemTicketsThunk({
         apiEndpoint: loginData.apiEndpoint!,
         apiToken: loginData.apiToken!,
       }),
-    )
-      .unwrap()
-      .then(() => {
-        set_redeeming(false);
-        handleRefresh();
-      })
-      .catch((e) => {
-        set_redeeming(false);
-        set_redeemErrors([
-          ...redeemErrors,
-          {
-            error: e.error,
-            status: e.status,
-          },
-        ]);
-      });
+    );
+    handleRefresh();
```

Comment:
this will refresh before the action is finished. do .unwrap and then .then and inside that then you can refresh.